### PR TITLE
Clean up docstrings and tests

### PR DESCRIPTION
**Pull request by jwhonce**
* Add unittest for exists()
* Add remove() on Resource and/or Manager for each resource
* Create SwarmNotImplementedError() to reduce duplicated strings
* Refactor RESTful error handling to reduce duplicated code
* Improve docstring formatting
* Fix volume RESTful calls
* Add integration testing for volumes
* Fix platform keyword parsing
* Add missing method return types

Signed-off-by: Jhon Honce <jhonce@redhat.com>

### CODE REVIEW

**Comment by jwhonce** [comment]
@mwhahaha PTAL

File: podman/api/client.py [hunk 1]
```diff
@@ -24,6 +24,44 @@
...
+    ) -> None:  # pylint: disable=arguments-differ
+        """Raises exception when Podman service reports one."""
+        if self.status_code < 400:
+            return
+
+        try:
+            body = self.json()
+            cause = body["cause"]
+            message = body["message"]
+        except json.decoder.JSONDecodeError:
```
Comment (mwhahaha) [inline podman/api/client.py:57]: Do we want to handle if there is no cause/message? The could be a KeyError thrown here

File: podman/api/client.py [hunk 2]
```diff
@@ -24,6 +24,44 @@
...
+
+    def __getattr__(self, item: str):
+        """Forward any query for an attribute not defined in this proxy class to wrapped class."""
+        return getattr(self._response, item)
+
+    def raise_for_status(
+        self, not_found: Type[APIError] = NotFound
+    ) -> None:  # pylint: disable=arguments-differ
+        """Raises exception when Podman service reports one."""
+        if self.status_code < 400:
```
Comment (mwhahaha) [inline podman/api/client.py:50]: < 300?  What happens if we get a redirect?

**Review by mwhahaha — APPROVED** [review]
Looks ok, have a few questions which could be followed up on in a later patch

File: podman/api/client.py (hunk 2 above)
Comment (jwhonce) [reply podman/api/client.py:50]: The requests library takes care of redirects.

File: podman/api/client.py (hunk 1 above)
Comment (jwhonce) [reply podman/api/client.py:57]: I'll add that.

**Review by mwhahaha — APPROVED** [review]
bors r+

**Pull request merged**